extracting_number_info.py

```diff
@@ -0,0 +1,24 @@
+MAJOR_COLORS = ["White", "Red", "Black", "Yellow", "Violet"]
+MINOR_COLORS = ["Blue", "Orange", "Green", "Brown", "Slate"]
+
+
+def color_pair_to_string(major_color, minor_color):
+  return f'{major_color} {minor_color}'
+  
+  
+  
+def get_pair_number_from_color(major_color, minor_color):
+  try:
+    major_index = MAJOR_COLORS.index(major_color)
+  except ValueError:
+    raise Exception('Major index out of range')
+  try:
+    minor_index = MINOR_COLORS.index(minor_color)
+  except ValueError:
+    raise Exception('Minor index out of range')
+  return major_index * len(MINOR_COLORS) + minor_index + 1
+  
+  
+def test_pair_to_number(major_color, minor_color, expected_pair_number):
```

Comment:
Production code and test code are in the same file. It is a good approach to keep them separately.
By doing so, we can test production code independently and effectively and also can exclude test code easily during shipment of software to field.